code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -2,10 +2,9 @@
 	if(M.a_intent == INTENT_DISARM)
 		if(!lying)
 			M.do_attack_animation(src, ATTACK_EFFECT_DISARM)
-			if(prob(85))
-				Stun(7)
-				step(src, get_dir(M,src))
-				spawn(5)
+			if(prob(55))
+				Stun(2)
+				spawn(2)
```

Comment:
not sure why you lowered the length spawn as well

Reply:
The spawn was a delay to make the robot move another tile after moving the first time from being shoved. They got shoved two tiles back.

As I changed it to be pushed back only one time, there was no reason to keep such a long delay.

Reply:
you don't actually need a spawn at all then, there's no reason for a delay if there's only the first push